src/eradiate/scenes/bsdfs/_hapke.py

```diff
@@ -0,0 +1,170 @@
+from ...units import unit_context_config as ucc
+
+
+@parse_docs
+@attrs.define(eq=False, slots=False)
+class HapkeBSDF(BSDF):
+    """
+    Hapke BSDF [``rtls``].
+
+    This BSDF implements the Hapke surface model as described in
+    :cite:`Hapke1984BidirectionalReflectanceSpectroscopy`. This highly flexible
+    and robust surface model allows for the characterisation of a sharp
+    back-scattering hot spot. The so-called Hapke model has been adapted to
+    several different use cases in the litterature, the version with 6
+    parameters implemented here is one of the most commonly used.
+
+    See Also
+    --------
+    :ref:`plugin-bsdf-rpv`
+    """
+
+    w: Spectrum = documented(
+        attrs.field(
+            default=None,
+            converter=spectrum_factory.converter("dimensionless"),
+            validator=[
+                attrs.validators.instance_of(Spectrum),
+                validators.has_quantity("dimensionless"),
+            ],
+        ),
+        doc="Single scattering albedo 'w'. " r"Must be in [0; 1]",
+        type=".Spectrum",
+        init_type=".Spectrum or dict or float",
+    )
+
+    b: Spectrum = documented(
+        attrs.field(
+            default=None,
+            converter=spectrum_factory.converter("dimensionless"),
+            validator=[
+                attrs.validators.instance_of(Spectrum),
+                validators.has_quantity("dimensionless"),
+            ],
+        ),
+        doc="Anisotropy parameter 'b' " r"Must be in [0; 1]",
+        type=".Spectrum",
+        init_type=".Spectrum or dict or float",
+    )
+
+    c: Spectrum | None = documented(
+        attrs.field(
+            default=None,
+            converter=spectrum_factory.converter("dimensionless"),
+            validator=[
+                attrs.validators.instance_of(Spectrum),
+                validators.has_quantity("dimensionless"),
+            ],
+        ),
+        doc="Scattering coefficient 'c'. " r"Must be in [0; 1]",
+        type=".Spectrum",
+        init_type=".Spectrum or dict or float",
+    )
+
+    theta: Spectrum = documented(
+        attrs.field(
+            default=0.183,
+            converter=spectrum_factory.converter("angle"),
+            validator=[
+                attrs.validators.instance_of(Spectrum),
+                validators.has_quantity("angle"),
+            ],
+        ),
+        doc="Photometric roughness 'theta'. Angle in degree" r"Must be in [0; 90]°",
+        type=".Spectrum",
+        init_type="quantity or float",
+    )
+
+    B_0: Spectrum = documented(
+        attrs.field(
+            default=None,
+            converter=spectrum_factory.converter("dimensionless"),
+            validator=[
+                attrs.validators.instance_of(Spectrum),
+                validators.has_quantity("dimensionless"),
+            ],
+        ),
+        doc="Shadow hiding opposition effect amplitude 'B_0'. " r"Must be in [0; 1]",
```

Comment:
Ditto

Reply:
It's indeed an artifact due to Black processing. It used to be on 2 lines